easybuild/easyblocks/generic/cargo.py

```diff
@@ -295,10 +313,18 @@ def extract_step(self):
                     src['finalpath'] = previous_source['finalpath']
                     continue
 
-            is_vendor_crate = src['name'] in vendor_crates
+            # Check if the source is a vendored crate and store the crate name
+            try:
+                crate_name = vendor_crates[src['name']][0]
+            except KeyError:
+                is_vendor_crate = False
+            else:
+                is_vendor_crate = True
+                src['crate_name'] = crate_name
```

Comment:
what about:
```suggestion
            # Check if the source is a vendored crate and store the crate name
            is_vendor_crate = src['name'] in vendor_crates
            if is_vendor_crate:
                src['crate_name'] = crate_name
```

Reply:
That is missing the `crate_name` retrieval.   
I wanted to stay true to the Python paradigm of "ask for forgiveness not permission" and avoid the double (quadruple) lookup by (not) having this in the last line: `src['crate_name'] = vendor_crates[src['name']][0]`